### DISCUSSION TITLE
Change appearance of code blocks

### DISCUSSION THREAD

**Discussion by LuisJoseSanchez** [Zettlr/Zettlr#2538] [Q&A]
Hi everyone,

I have some Java code and console command examples in my documents ¿Is there any way to change the styles of the code blocks? i.e. adding line numbers, setting a background color, setting the comments to a specific color, etc.

Thank you in advance!

**Comment by nathanlesage** [comment]
Coloring is easily possible using [Custom CSS](https://docs.zettlr.com/en/core/custom-css/). Adding line numbers unfortunately not (yet).

**Reply by LuisJoseSanchez** [reply]
Thank you @nathanlesage for your response. Maybe my question was not enough clear. As I see in the documentation linked in your message, you can personalize the app itself with custom CSS which is nice. But, what I want is to change the look of the code blocks (and eventually the rest of the document) in the pdf file generated.

I did some research and found that a TeX file with custom settings can be included before generate the pdf. Although I don't know much about TeX (almost nothing), I have tried to change the style of the code blocks in Java in my document with some templates that I have found on the internet but I have not succeeded.

Is there a simple TeX template for Zettler easy to customize?

PD: I use programs like Zettler to avoid TeX and LaTex :laughing:

**Reply by nathanlesage** [reply]
> in the pdf file generated.

Aaaaaaaaaaaaahhh, now I see! Sorry :D 

> Is there a simple TeX template for Zettler easy to customize?

Unfortunately not :( LaTeX is a PITA, and even Markdown can't completely shield you from that. So while Markdown makes it much easier to produce the text, you do have to customize at least one TeX template once ...

**Comment by krisdez** [comment]
Hello
Excuse my English which is not very correct (i'm french)

In my assets settings I modified the highlight-styles line with one of the following syntax coloring options (pygments, tango, espresso, zenburn, kate, monochrome, breezedark, haddock).

Then in my code blocks to have the lines numbered, I do like this
{ .html .numberLines}
and that numbers the lines in my code blocks.

I don't know if this will help but I'm using zettlr more and more and looking for interesting exports.
There are few docs in French and it's not easy when you don't know English.

Have a nice day
